libs/dal/src/repositories/environment/environment.repository.ts

```diff
@@ -87,4 +88,23 @@ export class EnvironmentRepository extends BaseRepository<EnvironmentDBModel, En
 
     return await this.getApiKeys(environmentId);
   }
+
+  async updateApiRateLimits(environmentId: string, apiRateLimits: Partial<IApiRateLimitMaximum>) {
+    await this.update(
+      {
+        _id: environmentId,
+      },
+      [
+        {
+          $set: {
+            apiRateLimits: {
+              $mergeObjects: ['$apiRateLimits', apiRateLimits],
+            },
+          },
+        },
+      ]
+    );
+
+    return await this.getApiKeys(environmentId);
```

Comment:
Do we need to return the api keys here?

Reply:
Good catch! This was accidentally left behind from a copy-paste. I've updated it to return the `this.update` result instead.